src/Symfony/Component/Config/Definition/Builder/ArrayNodeDefinition.php

```diff
@@ -372,7 +376,7 @@ protected function createNode()
                 $node->setKeyAttribute($this->key, $this->removeKeyItem);
             }
 
-            if (true === $this->atLeastOne) {
+            if (true === $this->atLeastOne || false === $this->allowEmptyValue) {
```

Comment:
I'd favor simplicity her, taking history into account, thus stick to `requiresAtLeastOneElement` and make the PR as small as possible. It's not like `requiresAtLeastOneElement` is a bad name.